jujugui/static/gui/src/app/views/topology/service.js

```diff
@@ -151,10 +155,32 @@ YUI.add('juju-topology-service', function(Y) {
       } else if (d.highlighted) {
         rerenderRelations = true;
       } else {
-        curr_node.attr({
-          'stroke': '#888888',
-          'stroke-width': 1
-        });
+        var units = d.units._items;
+        for (var i = 0; i < units.length; i++) {
```

Comment:
This could be a pretty big performance bottleneck in large environments. We should have a flag on the units modellist which indicates when a unit's agent state goes into error, pending, etc so that the multiple places around the app which require this information can use it. 

Could you take a look at this, and if it looks like too much work for this branch create a follow-up card. thx!